numpy/lib/function_base.py

```diff
@@ -1915,10 +1934,74 @@ def diff(a, n=1, axis=-1):
     slice1 = tuple(slice1)
     slice2 = tuple(slice2)
     if n > 1:
-        return diff(a[slice1]-a[slice2], n-1, axis=axis)
-    else:
+        return diff(a[slice1]-a[slice2], n-1, axis=axis, 
+            to_begin=to_begin, to_end=to_end)
+
+    # if nothing to add to either side, fast track the result
+    elif to_begin is None and to_end is None:
         return a[slice1]-a[slice2]
 
+    else:
+        # logic to handle to_end input
+        # l_end is the length of padding at the end
+        if to_end is None:
+            l_end = 0
+        else:
+            to_end = np.asanyarray(to_end)
+            if to_end.ndim == 0:
+                l_end = 1
+            elif to_end.ndim != a.ndim:
+                raise ValueError('to_end must have the same dimension as ary')
```

Comment:
ary was mistakenly copied from ediff1d.  Should this be "input_array", which is clearer name, or "a" to match the function signature?